### ISSUE TITLE
Error export ONNX in FP16

### ISSUE THREAD

**Issue by zha76** [CMPT756-A5-Org-Zeyu-Hu/MYC756PROJECT#13]
YOLOv5 Component
Export

Bug
When you exports ONNX with FP16 (half), there is an error and the .onnx file has an error:

python3 export.py --img {imgsize} --device 0 --weights {project}/{name}/weights/best.pt --include onnx --half --simplify

export: data=data/coco128.yaml, weights=['mot/bb_v5n2/weights/best.pt'], imgsz=[2048], batch_size=1, device=0, half=True, inplace=False, train=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=12, verbose=False, workspace=4, nms=False, agnostic_nms=False, topk_per_class=100, topk_all=100, iou_thres=0.45, conf_thres=0.25, include=['onnx']
YOLOv5 🚀 v6.0-237-gdc7e093 torch 1.10.1+cu113 CUDA:0 (NVIDIA GeForce RTX 3090, 24268MiB)

Fusing layers... 
Model Summary: 213 layers, 1760518 parameters, 0 gradients, 4.2 GFLOPs

PyTorch: starting from mot/bb_v5n2/weights/best.pt with output shape (1, 258048, 6) (5.8 MB)

ONNX: starting export with onnx 1.10.2...
ONNX: simplifying with onnx-simplifier 0.3.6...
**ONNX: simplifier failure: [ONNXRuntimeError] : 1 : FAIL : Type Error: Type parameter (T) of Optype (Concat) bound to different types (tensor(float) and tensor(float16) in node (Concat_307).**
ONNX: export success, saved as mot/bb_v5n2/weights/best.onnx (7.7 MB)
Environment
VoloV5 V6, RTX 3090,

**Comment by zha76** [comment]
fixed
